sched/scheduler/workGenerator_test.go

```diff
@@ -0,0 +1,54 @@
+package scheduler
+
+import (
+	"github.com/golang/mock/gomock"
+	"github.com/scootdev/scoot/saga"
+	"github.com/scootdev/scoot/sched"
+	"github.com/scootdev/scoot/sched/queue"
+	"testing"
+)
+
+func TestGenerateWork_ReadsAllItemsFromChannel(t *testing.T) {
```

Comment:
I'm confused why as well, this function doesn't even require a full queue, just a channel that has work on it.  And its an interface.  I just want to ensure Dequeue is called successfully which the Mock easily tests.  Dequeue doesn't currently work with what's checked into master as well.  

Same with the Scheduler Mock, super easy to test error cases by just telling the mock to return one.